cmd/ipfs/daemon.go

```diff
@@ -461,15 +471,25 @@ func serveHTTPApi(req *cmds.Request, cctx *oldcmds.Context) (<-chan error, error
 		return nil, fmt.Errorf("serveHTTPApi: ConstructNode() failed: %s", err)
 	}
 
-	if err := node.Repo.SetAPIAddr(apiMaddr); err != nil {
+	if err := node.Repo.SetAPIAddr(listeners[0].Multiaddr()); err != nil {
```

Comment:
do we want to connect the repo to only one of the addresses? Should it always be the first?

Reply:
So, adding support for multiple addrs here would require changing the `~/.ipfs/api` file format (or, more likely, introducing a *second* API file with support for multiple addresses). Choosing the *first* address is probably reasonable for now. Eventually, we may want to consider preferring a loopback address, or something like that, but we can revisit this later.